src/category_theory/preadditive/single_obj.lean

```diff
@@ -0,0 +1,27 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.preadditive.default
+import category_theory.single_obj
+
+/-!
+# `single_obj α` is preadditive when `α` is a ring.
+
+-/
+
+namespace category_theory
+
+variables {α : Type*} [ring α]
+
+instance (X Y : single_obj α) : add_comm_group (X ⟶ Y) :=
+by { change add_comm_group α, apply_instance, }
+
+instance : preadditive (single_obj α) :=
+{ add_comp' := λ _ _ _ f f' g, mul_add g f f',
+  comp_add' := λ _ _ _ f g g', add_mul g g' f, }
+
+-- TODO once we have preadditive functors, define `PreAdditiveCat`, and `Ring ⥤ PreAdditiveCat`.
```

Comment:
We have additive functors in category_theory.functor.additive. Do you mean something different by a "preadditive functor"?

Reply:
Oops, I wrote this before I'd caught up on what you'd added.
```suggestion
-- TODO define `PreAddCat` (with additive functors as morphisms), and `Ring ⥤ PreAdditiveCat`.
```